### PR TITLE
A few small Nomad Pack consumption adjustments

### PR DESCRIPTION
**Pull request by BrandonRomano**
## Description

- Allows for `pack.name` to be overridden via `integration.name`
- Removes the first line if it's an H1 in the README

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/integrations-hcl/strategies/nomad-pack/load_pack_directory.ts [hunk 1]
```diff
@@ -42,13 +42,28 @@ export async function loadNomadPackIntegrationDirectory(
...
+  const readmeLines = rawReadmeContent.split('\n')
+  const h1Regex = /^# .*/g
+  if (h1Regex.test(readmeLines[0])) {
+    readmeLines.shift()
+    // The next line is likely an empty line, so let's remove that
+    if (readmeLines[0] === '') {
+      readmeLines.shift()
+    }
+  }
+  const readmeContent = readmeLines.join('\n')
```
Comment (BrandonRomano) [inline packages/integrations-hcl/strategies/nomad-pack/load_pack_directory.ts:60]: https://raw.githubusercontent.com/hashicorp/nomad-pack-community-registry/main/packs/aws_efs_csi/README.md

Here's a concrete README that will be processed which will benefit from this parsing logic.

File: packages/integrations-hcl/strategies/nomad-pack/load_pack_directory.ts [hunk 2]
```diff
@@ -42,13 +42,28 @@ export async function loadNomadPackIntegrationDirectory(
...
-  const readmeContent: string = fs.readFileSync(readmeFile, 'utf8')
+
+  // Read the README
+  const rawReadmeContent: string = fs.readFileSync(readmeFile, 'utf8')
+
+  // If the first line is a H1, remove it. Keeping it will create a
+  // duplicate title in our content on the website.
+  const readmeLines = rawReadmeContent.split('\n')
+  const h1Regex = /^# .*/g
+  if (h1Regex.test(readmeLines[0])) {
```
Comment (thiskevinwang) [inline packages/integrations-hcl/strategies/nomad-pack/load_pack_directory.ts:53]: Suggestion, mostly for ourselves. Add a log statement to for this conditional action we're taking
... Something like: `console.log("Removing leading H1 from README content...")`

File: packages/integrations-hcl/strategies/nomad-pack/load_pack_directory.ts [hunk 3]
```diff
@@ -42,13 +42,28 @@ export async function loadNomadPackIntegrationDirectory(
...
+    }
+  }
+  const readmeContent = readmeLines.join('\n')
 
   return {
     id: integrationID,
     product: integrationProductSlug,
     identifier: integrationStanza.identifier,
-    name: packStanza.name,
+    name: integrationStanza.name ? integrationStanza.name : packStanza.name,
```
Comment (thiskevinwang) [inline packages/integrations-hcl/strategies/nomad-pack/load_pack_directory.ts:66]: 👍

**Review by thiskevinwang — APPROVED** [review]
Both these changes look good to me! Both good calls 😄

**Pull request merged**